File: PickingNumbers.js
'use strict';

const fs = require('fs');

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', function(inputStdin) {
    inputString += inputStdin;
});

process.stdin.on('end', function() {
    inputString = inputString.split('\n');

    main();
});

function readLine() {
    return inputString[currentLine++];
}

/*
 * Complete the 'pickingNumbers' function below.
 *
 * The function is expected to return an INTEGER.
 * The function accepts INTEGER_ARRAY a as parameter.
 */

function pickingNumbers(a) {
    a.sort((a, b) => a - b);
    let minValue = a[0];
    let len = 1;
    let maxLen = 1;
    for (let i = 1; i < a.length; i++) {
        const diff = a[i] - minValue;
        if (diff <= 1) {
            len++;
        } else {
            if (len > maxLen) {
                maxLen = len;
            }
            len = 1;
            minValue = a[i];
        }
    }
    if (len > maxLen) {
        maxLen = len;
    }
    return maxLen;
}

function main() {
    const ws = fs.createWriteStream(process.env.OUTPUT_PATH);

    const n = parseInt(readLine().trim(), 10);

    const a = readLine().replace(/\s+$/g, '').split(' ').map(aTemp => parseInt(aTemp, 10));

    const result = pickingNumbers(a);

    ws.write(result + '\n');

    ws.end();
}

File: PickingNumbers.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.function.*;
import java.util.regex.*;
import java.util.stream.*;
import static java.util.stream.Collectors.joining;
import static java.util.stream.Collectors.toList;

class Result
{
    public static int pickingNumbers(List<Integer> a)
    {
        int MAX_VALUE = 99;
        
        int[] freq = new int[MAX_VALUE + 1];
        for (int num: a)
        {
            freq[num]++;
        }
        
        int maxLen = 0;
        for (int i = 1; i < MAX_VALUE; i++)
        {
            int currentLen = freq[i] + freq[i + 1];
            maxLen = Math.max(maxLen, currentLen);
        }
        
        return maxLen;
    }
}

public class Solution {
    public static void main(String[] args) throws IOException {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(System.getenv("OUTPUT_PATH")));

        int n = Integer.parseInt(bufferedReader.readLine().trim());

        List<Integer> a = Stream.of(bufferedReader.readLine().replaceAll("\\s+$", "").split(" "))
            .map(Integer::parseInt)
            .collect(toList());

        int result = Result.pickingNumbers(a);

        bufferedWriter.write(String.valueOf(result));
        bufferedWriter.newLine();

        bufferedReader.close();
        bufferedWriter.close();
    }
}
